### DISCUSSION TITLE
前端发送到Aria2成功，可以下载到Aria2设定的目录下，但是离线下载到特定目录，无法在Alist下创建临时指定目录

### DISCUSSION THREAD

**Discussion by rocksea1973** [AlistGo/alist#3166] [Q&A] [answered]
例如：下载路径: /opt/alist/data/temp/aria2/d26c825c-2782-4c1d-a6c3-29c0919fa0a5
不知道哪里需要设置一下

**Comment by anwen-anyi** [comment, marked as answer]
config.json 文件里面的 `  "temp_dir": "data/temp",`这个控制的临时文件夹路径。

**Reply by rocksea1973** [reply]
多谢回复，我感觉是需要alist授权给aria2读写的权限，不知道为什么Aria2无法在alist目录下创建这个临时目录

**Reply by rocksea1973** [reply]
试着修改了一下路径，下载成功了，但是传送到指定路径失败了，failed to open file。。。no such file or directory

**Reply by rocksea1973** [reply]
aria2那边显示如下：

01/28 14:00:15 [NOTICE] Serialized session to '/config/aria2.session' successfully.
[#ac7b24 13MiB/17MiB(73%) CN:2 DL:5.5MiB]                                                                                                            
01/28 14:00:21 [NOTICE] Download complete: /tmp/temp/aria2/b513a4fc-a718-4b60-afbb-0ab483fd34df/04. 我没有骗你.flac
                                                                                                                                                     
01/28 14:00:22 [NOTICE] Serialized session to '/config/aria2.session' successfully.
01/28 14:00:22 [INFO] .aria2 file does not exist.
01/28 14:00:22 [INFO] General download task, skipped delete .torrent file.
01/28 14:00:22 [INFO] Deleting empty directory ...

alist 这边显示如下：

transfer /tmp/temp/aria2/b513a4fc-a718-4b60-afbb-0ab483fd34df/04. 我没有骗你.flac to [/253A-Music](/)
错误
failed to open file /tmp/temp/aria2/b513a4fc-a718-4b60-afbb-0ab483fd34df/04. 我没有骗你.flac: open /tmp/temp/aria2/b513a4fc-a718-4b60-afbb-0ab483fd34df/04. 我没有骗你.flac: no such file or directory

**Reply by anwen-anyi** [reply]
`to /253A-Music` 这个路径没有？    这个看起来像是根目录诶？根目录能上传文件吗？

**Reply by rocksea1973** [reply]
这个目录是我通过SMB挂载的NAS本地共享文件夹，但是看两边的过程，似乎还没有到传输这一步，好像aria2下载显示成功之后，不知道什么原因就给自动删除了，然后alist就没有找到文件去传输，我用的是p3terx的Aria2 Pro docker

**Reply by anwen-anyi** [reply]
emmm 这就不太懂了...QAQ

**Reply by rocksea1973** [reply]
谢谢
